W-17017575-insights-hyperforce-duke (#314)
mulesoft/docs-monitoring: modules/ROOT/pages/anypoint-insights.adoc

//= Explore Entities with Anypoint Insights
= Monitor Apps and APIs Across Deployments with Anypoint Insights
:page-notice-banner-message: Only Japan (JA) and Canada (CA) regions provide Anypoint Insights. No other Anypoint Monitoring features are offered in these regions at this time.

Monitor performance data and issues across your CloudHub 2.0 and Runtime Fabric deployments (_entities_) with Anypoint Insights, in Anypoint Monitoring. 

image::insights.png["Anypoint Insights UI"]

//TODO_TODO_TODO: 24 hours for now
Identify and troubleshoot up to 24 hours of data on your deployments with:

> * Aggregated metrics across your deployments.

Author: ```suggestion
* Metrics across multiple deployments
```

+
Monitor response times, error volume, and request volume. Use filters to narrow your results. 
* Metrics on individual apps and APIs.
+
Review requests, latency, errors, and JVM data.

For data collection details, see <<data-collection>>.

[[prereqs]]
== Before You Begin

Required prerequisites:

* An Anypoint Platform account in a region that supports Anypoint Insights

** {empty}
include::partial$include-nav-entities.adoc[tags="anypoint-regions-hyperforce"] 

** {empty}
include::partial$include-nav-entities.adoc[tags="anypoint-regions-useu"] 
(Beta release for Advanced and Titanium tier plans only at this time)

* The *Monitoring Viewer* permission to Anypoint Monitoring and the permissions required for viewing apps and APIs.
+
See xref:monitoring::am-permissions.adoc[] and xref:access-management::permissions-by-product.adoc[].
* A supported version of Mule Runtime engine.
+
See xref:release-notes::mule-runtime/mule-esb.adoc[].

> 

Author: TODO: Add to BYB -> Need deployed app and/or apis

Author: done

== Open Anypoint Insights

//include::partial$include-nav-steps-no-permissions.adoc[]

. Meet the prerequisites in <<prereqs>>. 
. Log in to your Anypoint Platform region that supports Anypoint Insights:
+

> include::partial$include-nav-entities.adoc[tags="supported-regions"] 

Author: TODO: verify supported-regions

Author: outdated

//Nav step
include::partial$include-nav-entities.adoc[tags="open-insights"]
+

> *Anypoint Insights* provides separate sets of charts for your deployment environments, such as Design, Production, and Sandbox. You can view and filter chart data across your deployments within the selected environment, and you can open more detailed data on any deployment listed in the *Entities List*. 

Author: TODO: mention environments in intro

Author: outdated

[[entities]]
include::partial$include-entities.adoc[tags="entity-definition"]

[[dashboard-overview]]
//don't change order of numbers in graphic or calloutlist
image::insights.png["Anypoint Insights UI"]

[calloutlist]
. Dashboard that provides an overview of metrics across of your deployed Mule apps and APIs
. Environment selector for your deployments (such as Sandbox or Production)
. <<filter, Filters>> to narrow the data in charts
. Performance <<charts-performance, metrics and trends>> across your deployed Mule apps and APIs
. <<entities-list, List of each Mule app and API>> that you can click to display <<charts-detailed, detailed metrics>> on a specific deployment
. Selector for the time range covered by the charts 
+
{selector-all-dashboards}
. Selector for auto-refresh interval, displayed when auto-refresh is enabled 
+
{selector-all-dashboards}

[[prereqs]]
== Before You Begin

Required prerequisites:

* An Anypoint Platform account in a region that supports Anypoint Insights

** {empty}
include::partial$include-entities.adoc[tags="anypoint-regions-hyperforce"] 

** {empty}
include::partial$include-entities.adoc[tags="anypoint-regions-useu"]

> * A Mule app or API running on a supported version of Mule Runtime engine (see xref:release-notes::mule-runtime/mule-esb.adoc[]) and deployed to a supported target:

Author: Question @sathishpaulleo: Is the API is a Mule Gateway API, or can it be another type, like Flex Gateway?

Author: from Sathish: can be either (just these)

> * A Mule app or API implementation running on a supported version of Mule Runtime engine (see xref:release-notes::mule-runtime/mule-esb.adoc[]) and deployed to a supported target:

Author: ```suggestion
* A Mule app or API running on a supported version of Mule Runtime engine (see xref:release-notes::mule-runtime/mule-esb.adoc[]) and deployed to a supported target:
```

. Select the file format for your metrics:

* *Download as CSV*
* *Download as XLS*

//H2: REFERENCES
== Insights Chart Reference

Understand the monitoring data in Anypoint Insights charts:

* <<charts-performance>>
* <<entities-list>>
* <<entity-metrics-apps>>
* <<entity-metrics-apis>>

//H3: PERFORMANCE
[[charts-performance]]
=== Performance Metrics and Trends

*Performance Metrics* and *Performance Trends* provide insights across your deployments:

* <<performance-metrics-overview>>
* <<performance-trends-overview>>

//  H4: Performance Metrics
[[performance-metrics-overview]]
==== Performance Metrics

Monitor high-level performance metrics across your deployments. 

include::partial$include-insights-charts.adoc[tags="data-scope"]

Total Entities::
Total number of entities

Entities with Errors::
Total number of entities that have received errors

* For API implementations, this metric counts HTTP status codes that identify errors (for example, `404` or `502`). 
* For Mule apps, this metric counts response codes that indicate an error (for example, `FAILED`).

Total Requests::
Total number of requests to the listed entities

Error Rate:: 

> For entities with request errors, the percentage equal to the total number of request errors  divided by the total number of requests (with and without errors)

Author: ```suggestion
For entities with errors, the percentage equal to the total number of errors divided by the total number of requests
```

+

> For example, assume that 1 out of 5 total requests to API A produces an error and that 2 requests out of 5 to Application B produces an error. The error rate is 3/10, or 30%. 

Author: TODO: pls verify description @sathishpaulleo 

Missed in last meeting.

//
// H4: Performance Trends
[[performance-trends-overview]]
==== Performance Trends

Monitor high-level performance over time across your deployments. 

include::partial$include-insights-charts.adoc[tags="data-scope"]

> //TODO_TODO: slow -> Slow
> //TODO_TODO: is the percentile calculated in the normal way? are we saying x% are faster or slower?
> Entities with Slow Response Times::
> Response times, in milliseconds, of the entities with the slowest response times
> +
> The metric uses percentile values for response times (calculated from highest to lowest times) to identify entities with the slowest responses. You can select the *p99* (default), *p95*, *p90*, or *p75*.

Author: Pls address todo and verify description, @sathishpaulleo 
Not covered in last meeting.

```suggestion
//TODO_TODO: slow -> Slow
//TODO_TODO: is the percentile calculated in the normal way? are we saying x% are faster or slower?
Entities with Slow Response Times::
Response times, in milliseconds, of the entities with the slowest response times
+
The metric uses percentile values for response times to identify entities with the slowest responses. You can select the *p99* (default), *p95*, *p90*, or *p75*.
```

= Monitor Deployments with Anypoint Insights
:page-notice-banner-message: For Japan (JP) and Canada (CA) instances of Anypoint Platform, Anypoint Insights is available in the Starter package. For US and EU instances, Anypoint Insights remains a Beta release available in the Advanced package only at this time.
:selector-all-dashboards: This selector is available to all Insights dashboards.
:download-chart-data: You can download most chart data to CSV or XSL.

Monitor performance data and issues across your deployed Mule apps and API implementations (_entities_) with Anypoint Insights, in Anypoint Monitoring. 

Identify and troubleshoot data on your deployments with:

* Metrics across multiple Mule app and API deployments
+
//TODO: VERIFY 7 DAYS (SATHISH)

> Get up to the last 7 days of metrics on response times, error volume, and request volume. Use filters to narrow your results. {download-chart-data}

Author: ```suggestion
//TODO: VERIFY 7 DAYS (SATHISH)
Get up to the last 7 days of metrics on response times, error volume, and request volume. Use filters to narrow your results. {download-chart-data}
```
@sathishpaulleo

Author: answer: 24 hours for starter, 7 days for advanced

* Metrics on individual Mule app and API deployments
+

> //TODO: VERIFY 30 DAYS (SATHISH)
> Get up to the last 30 days of metrics on requests, latency, errors, JVM data, performance, and client apps. {download-chart-data}

Author: ```suggestion
//TODO: VERIFY 30 DAYS (SATHISH)
Get up to the last 30 days of metrics on requests, latency, errors, JVM data, performance, and client apps. {download-chart-data}
```
@sathishpaulleo

Author: yes

[calloutlist]
. Dashboard that provides an overview of metrics across of your deployed Mule apps and APIs
. Environment selector for your deployments (such as Sandbox or Production)
. <<filter, Filters>> to narrow the data in charts
. Performance <<charts-performance, metrics and trends>> across your deployed Mule apps and APIs
. <<entities-list, List of each Mule app and API>> that you can click to display <<charts-detailed, detailed metrics>> on a specific deployment
. Selector for the time range covered by the charts 
+
{selector-all-dashboards}
. Selector for auto-refresh interval, displayed when auto-refresh is enabled 
+
{selector-all-dashboards}

[[prereqs]]
== Before You Begin

Required prerequisites:

* An Anypoint Platform account in a region that supports Anypoint Insights

** {empty}
include::partial$include-nav-entities.adoc[tags="anypoint-regions-hyperforce"] 

** {empty}
include::partial$include-nav-entities.adoc[tags="anypoint-regions-useu"] 
(Beta release for Advanced and Titanium tier plans only at this time)

* A running Mule app or Mule Gateway API that is deployed to a supported target:

** Runtime Fabric
** CloudHub 2.0

> //TODO_TODO_TODO: hybrid not supported? CH 1 not supported? 

Author: ```suggestion
```

Author: US and EU advanced & titanium, yes CH 1 & Hybrid

* CloudHub 1.0 (US and EU only)
* Hybrid (US and EU only)

Author: done

* The *Monitoring Viewer* permission to Anypoint Monitoring and the permissions required for viewing apps and APIs (see xref:monitoring::am-permissions.adoc[])
* A supported version of Mule Runtime engine (see xref:release-notes::mule-runtime/mule-esb.adoc[])

[[monitor-deployments]]
== Monitor Your Deployments

Monitor Mule apps and APIs within a selected business group and environment.

> //To review metrics:

Author: ```suggestion
```

Author: done

. Meet the prerequisites in <<prereqs>>. 
. Log in to Anypoint Platform, and navigate to Anypoint Monitoring. 
//Nav step
include::partial$include-nav-entities.adoc[tags="open-insights"]
. In Anypoint Insights, set your environment, such as Design, Production, and Sandbox. 
+
For guidance, see <<change-environment>>.
. Set the time period to capture in your charts.
+
For guidance, see <<filter-by-time>>.
. Filter the deployment data to display in the charts, as needed, to focus on areas of interest.
+
For guidance, see <<filter-by-data>>.
. To get additional metrics on a specific app or API:
.. Find and click its name in *Entities List*.

> .. Click *View Entity Metrics*, for example:

Author: Could shorten this to *View Metrics* 
```suggestion
.. Click *View Entity Metrics*, for example:
```
@sathishpaulleo

Author: Also, need to conform capitalization of labels like these, either capitalizing the second word or not everywhere unless the word must always be capitalized (such as URL, API, etc.):

Entity type
Business group
Environment
Request Volume
Target

@sathishpaulleo

Author: It's interesting that we call CloudHub a Target here (and not a Deployment Type), for example:
Target: cloudhub-us-east-1

@sathishpaulleo

Author: Url - URL, Api -> API

+
image::insights-entity-clicked.png["Overview of a deployment metrics"]
.. Review metrics in the charts for the entity, for example: 
+
[[charts-detailed]]

> image::insights-deployment-charts.png["Entity dashboard for a deployed app or API"]

Author: ```suggestion
image::insights-deployment-charts.png["Dashboard for a deployed app or API"]
```

Author: done

* <<filter-by-data>>
* <<filter-by-time>>

//  H3: FILTER DEPLOYMENT DATA
[[filter-by-data]]
=== Filter Deployment Data

Use one or more filters to narrow the data in your charts. The filters support operators `=` and `IN` (for any matches to a list of items). Multiple filter settings are allowed. For example, this filter returns Mule apps deployed to CloudHub 2.0: 

----
entity_type = 'APP' deployment_type = 'CLOUDHUB2'
---- 

This table lists supported filters:

[%header,cols="10a,80a,10a"]
|===
| Filter | Operators | Description

|`deployment_type`
|`=`, `IN`
|Filter by deployment target, CloudHub 2.0 (*CLOUDHUB2*) or Runtime Fabric (*RTF*), for example:
----
deployment_type IN ['CLOUDHUB2', 'RTF']
----

|`entity_type`
|`=`, `IN`
|Filter by deployment type (*APP* or *API*), for example:
----
entity_type = 'APP'`
----

|`entity`
|`=`, `IN`
|Filter by specific app or API deployments, for example:
----
entity IN ['99aea888-77de-6e06-5c55-d1234de56789', '09f99999-888d-7b77-6da5-bebd43210fc1']
----

|`entity_response_status`
|`=`
|Filter by the response code (*OK* or *FAILED*), for example:
----
entity_response_status = 'OK'

> ----

Author: ```suggestion
|`entity_response_status`
|`=`
|Filter by the response code (*OK* or *FAILED*), for example:
----
entity_response_status = 'OK'
----
```
TODO: remove @dukesphere

Author: done

include::partial$include-insights-charts.adoc[tags="data-scope"]

Entities with Errors::
Count of entities that have an error that matches your current search conditions in the selected date and time range.
include::partial$include-insights-charts.adoc[tags="data-scope"] 
+
Errors are dependent on the type of deployment. 

* For API implementations, an error is based on the HTTP status code (for example, `404` or `502`). 
* For Mule apps, an error is based on the response status (for example, `FAILED`).

Total Requests::
Total request volume for entities that match your current search conditions in the selected date and time range.
include::partial$include-insights-charts.adoc[tags="data-scope"] 

Error Rate:: 
Error rate for entities that match your current search conditions in the selected date and time range. The error rate is calculated using the total number of entity errors divided by the total number of requests for those entities.
include::partial$include-insights-charts.adoc[tags="data-scope"]

//
// H4: Performance Trends
[[performance-trends-overview]]
==== Performance Trends

> Entities with Slow Response Times:: 

Author: UI currently says Entities with slow Response Times
slow -> Slow
@sathishpaulleo

Author: ```suggestion
Entities with Slow Response Times:: 
```

Author: UI currently says Entities with slow Response Times
slow -> Slow

Response time of the top entities that match your current search conditions in the selected date and time range. The top entities are calculated based on the percentile values of their response times (highest response time to lowest response time). The chart uses `P99` as the percentile index by default; however, you can select `P95`, `P90`, or `P75` to use as the percentile index.
include::partial$include-insights-charts.adoc[tags="data-scope"]

Error Volume:: 
Total number of entities that match your current search conditions in the selected date and time range.
include::partial$include-insights-charts.adoc[tags="data-scope"]
+
Errors are dependent on the context of the entity. For API entities, an error is based on the HTTP status code (for example, `4xx` or `5xx`) and for application entities, an error is based on the response status (for example, `FAILED`)
Request Volume:: 
The total number of request to entities that match your current search conditions in the selected date and time range. 

//
//H3: ENTITIES LIST
[[entities-list]]

> === Entities List

Author: Suggest removing "List" 
UX is generally to avoid unnecessary terms like this (self-identification that is obvious)

@sathishpaulleo

Author: ```suggestion
=== Entities List
```

Anypoint Insights provides a list of deployed Mule apps and APIs (_entities_) that match the selected time period and any query <<filters, filters>>:

Entity Name:: 
Name of the deployed entity

//TODO_TODO_TODO: Mule Gateway APIs only or other API instance?

> //TODO: SHOULD BE "API" not "Api"

Author: @sathishpaulleo 
API is always fully capitalized

Please change "Api" to "API"

Entity Type:: 

> Type of entity deployed, such as *API* or *Application*

Author: ```suggestion
//TODO_TODO_TODO: Mule Gateway APIs only or other API instance?
//TODO: SHOULD BE "API" not "Api"
Entity Type:: 
Type of entity deployed, such as *API* or *Application*
```

@sathishpaulleo
API is always fully capitalized in a context like this

Please change "Api" to "API"

Request Volume::
Total number of requests to this entity

Response Time (p99):: 
Average response time, in milliseconds, for 99% of the requests

> +

Author: Let's verify my understanding of how the pN values are calculated. @sathishpaulleo

> Calculation of this metric starts from the lowest response time value, so the time for the remaining, uncounted 1% of responses is likely to be higher than the p99 response time.

Author: ```suggestion
Response Time (p99):: 
Average response time, in milliseconds, for 99% of the requests
+
Calculation of this metric starts from the lowest response time value, so the time for the remaining, uncounted 1% of responses is likely to be higher than the p99 response time.
```
Please verify my understanding of how the pN values are calculated. @sathishpaulleo

Error Rate:: 
Percentage of requests that resulted in errors 

> Deployment Type:: 

Author: I'm not clear why we call this a Deployment Type

> Deployment Type:: 

Author: ```suggestion
Deployment Type:: 
```
Not clear why we call this a Deployment Type

We've used language like "Target" elsewhere

Author: this will remain the same (corresponds to filter deployment_type

.. In Access Management *Settings*, select a business group to use as the default, for example:
+
image::insights-default-selector.png["Default selector in Access Management"]
.. Return to Anypoint Monitoring to check for the new default setting.
+
Refresh the page, if necessary. 

[[download-chart-data]]
== Download Chart Data

After opening a set of charts in Anypoint Insights, you can download metrics from charts that provide a download menu from a vertical ellipsis on the top-right corner.

. Click the vertical ellipsis on a chart. 
. Select the file format for your metrics:

* *Download as CSV*
* *Download as XLS*

//H2: REFERENCES
== Insights Chart Reference

Understand the monitoring data in Anypoint Insights charts:

* <<charts-performance>>
* <<entities-list>>
* <<entity-metrics-apps>>
* <<entity-metrics-apis>>

include::partial$include-entities.adoc[tags="entity-definition"]

//H3: PERFORMANCE
[[charts-performance]]
=== Performance Metrics and Trends

*Performance Metrics* and *Performance Trends* provide insights across your entities:

* <<performance-metrics-overview>>
* <<performance-trends-overview>>

> //  H4: Performance Metrics
> [[performance-metrics-overview]]
> ==== Performance Metrics
> 

Author: ```suggestion
//  H4: Performance Metrics
[[performance-metrics-overview]]
==== Performance Metrics

```
@sathishpaulleo Pls review the descriptions of these charts

Monitor high-level performance metrics across your entities. 

include::partial$include-entities.adoc[tags="entity-definition"]

include::partial$include-insights-charts.adoc[tags="data-scope"]

Total Entities::
Total number of entities

Entities with Errors::
Total number of entities that have received errors

* For API implementations, this metric counts HTTP status codes that identify errors (for example, `404` or `502`). 
* For Mule apps, this metric counts response codes that indicate an error (for example, `FAILED`).

Total Requests::
Total number of requests to the listed entities

Error Rate:: 
For entities with request errors, the percentage equal to the total number of request errors divided by the total number of requests
+
For example, assume that 1 out of 5 total requests to _API A_ produces an error and that 2 requests out of 5 to _App B_ produces an error. The error rate is 3/10, or 30%. 

//

> // H4: Performance Trends
> [[performance-trends-overview]]
> ==== Performance Trends

Author: ```suggestion
// H4: Performance Trends
[[performance-trends-overview]]
==== Performance Trends
```
@sathishpaulleo Pls review the descriptions of these charts

Monitor high-level performance over time across your entities. 

include::partial$include-entities.adoc[tags="entity-definition"]

include::partial$include-insights-charts.adoc[tags="data-scope"]

> //TODO_TODO: slow -> Slow
> //TODO_TODO: is the percentile calculated in the normal way? are we saying x% are faster or slower?
> Entities with Slow Response Times::
> Response times, in milliseconds, of the entities with the slowest response times
> +
> The metric uses percentile values for response times (calculated from highest to lowest times) to identify entities with the slowest responses. You can select the *p99* (default), *p95*, *p90*, or *p75*.

Author: ```suggestion
//TODO_TODO: slow -> Slow
//TODO_TODO: is the percentile calculated in the normal way? are we saying x% are faster or slower?
Entities with Slow Response Times::
Response times, in milliseconds, of the entities with the slowest response times
+
The metric uses percentile values for response times (calculated from highest to lowest times) to identify entities with the slowest responses. You can select the *p99* (default), *p95*, *p90*, or *p75*.
```
@sathishpaulleo

= Monitoring Metrics from Anypoint Insights

> :page-notice-banner-message: For Japan Cloud and Canada Cloud instances of Anypoint Platform, Anypoint Insights is available in the Starter package. For US and EU clouds, Anypoint Insights remains a Beta release available in the Advanced and Titanium tiers at this time.

Reviewer: Remove "at this time"?

Author: intentional

:selector-all-dashboards: This selector is available to all Insights dashboards.
:download-chart-data: You can download most chart data to CSV or XSL.

//feature tier:
include::reuse::partial$billing/pricing.adoc[tag=featureUsageTitReqMonitoring]

> Monitor performance data and issues across your deployed Mule apps and API implementations (_entities_) with Anypoint Insights, in Anypoint Monitoring: 

Reviewer: ```suggestion
Monitor performance data and issues across your deployed Mule apps and API implementations (_entities_) with Anypoint Insights in Anypoint Monitoring: 
```

> * Check metrics across multiple Mule apps and API implementations

Reviewer: ```suggestion
* Check metrics across multiple Mule apps and API implementations.
```

+
Get up to the last 7 days (Advanced package) or 24 hours (Starter package) of high-level metrics on requests, errors, and responses. Use filters to narrow your results. {download-chart-data}

> * Check metrics on individual Mule apps and API implementations

Reviewer: ```suggestion
* Check metrics on individual Mule apps and API implementations.
```

[[entities]]
include::partial$include-entities.adoc[tags="entity-definition"]

[[dashboard-overview]]
//don't change order of numbers in graphic or calloutlist
image::insights.png["Anypoint Insights UI"]

[calloutlist]
. Dashboard that provides an overview of metrics across of your deployed Mule apps and APIs
. Environment selector for your deployments (such as Sandbox or Production)
. <<filter, Filters>> to narrow the data in charts
. Performance <<charts-performance, metrics and trends>> across your deployed Mule apps and APIs
. <<entities-list, List of each Mule app and API>> that you can click to display <<charts-detailed, detailed metrics>> on a specific deployment
. Selector for the time range covered by the charts 
+
{selector-all-dashboards}
. Selector for auto-refresh interval, displayed when auto-refresh is enabled 
+
{selector-all-dashboards}

[[prereqs]]
== Before You Begin

Required prerequisites:

* An Anypoint Platform account in a control plane that supports Anypoint Insights:

** {empty}
include::partial$include-entities.adoc[tags="anypoint-regions-hyperforce"] 

** {empty}
include::partial$include-entities.adoc[tags="anypoint-regions-useu"]

> * A Mule app or API (xref:gateway-home::index.adoc[Mule Gateway or Flex Gateway API]) running on a supported version of Mule Runtime engine (see xref:release-notes::mule-runtime/mule-esb.adoc[]) and deployed to a supported target:

Reviewer: ```suggestion
* A Mule app or API (xref:gateway-home::index.adoc[Mule Gateway or Flex Gateway API]) running on a supported version of Mule runtime engine (see xref:release-notes::mule-runtime/mule-esb.adoc[]) and deployed to a supported target:
```

** CloudHub 2.0
//TEMP INFO: Hyperforce 12/24 (W-17225983)
** CloudHub (for US and EU clouds, in Advanced and Titanium tiers only)
//TEMP INFO: Hyperforce 12/24 (W-17225983)
** Hybrid (for US and EU clouds, in Advanced and Titanium tiers only)
** Runtime Fabric

* For Canada Cloud and Japan Cloud: 

** Enable Mule runtime access to Anypoint Platform so that you can display entity metrics in Anypoint Monitoring.
+
For guidance, see xref:hyperforce::configure-mule-hyperforce.adoc[].
** For information about Runtime Fabric feature support in these clouds, see xref:hyperforce::index.adoc#features-supported-on-hyperforce[Features Supported on Hyperforce].

> * The *Monitoring Viewer* permission to Anypoint Monitoring and the permissions required for viewing apps and APIs (see xref:monitoring::am-permissions.adoc[])

Reviewer: ```suggestion
* The Monitoring Viewer permission to Anypoint Monitoring and the permissions required for viewing apps and APIs (see xref:monitoring::am-permissions.adoc[])
```

[[monitor-deployments]]
== Monitor Your Deployments

Monitor Mule apps and APIs within a selected business group and environment.

. Meet the prerequisites in <<prereqs>>. 
. Log in to Anypoint Platform, and navigate to Anypoint Monitoring. 
//Nav step
include::partial$include-entities.adoc[tags="open-insights"]

> . In Anypoint Insights, set your environment, such as Design, Production, and Sandbox. 

Reviewer: ```suggestion
. In Anypoint Insights, set your environment, such as Design, Production, or Sandbox. 
```

+

> For guidance, see <<change-environment>>.

Reviewer: For these "For guidance", you could consider using inline links to streamline the list:
"<<change-environment,set your environment>>"

Author: @dukesphere [W-17361464](https://gus.lightning.force.com/a07EE000026JsovYAC) future

Author: The more I think about this, I'd rather people not click to get how-to info on a step unless they decide they need it. I think they're more likely to click the link to the info if that link is part of the step vs. in a sentence below. (They might think they need to click it for some reason if it's embedded in the step.)

. To capture deployments for another business group you belong to in the *Entities List*, select the business group.
+
For guidance, see <<change-bg>>.
. Set the time period to capture in your charts.
+
For guidance, see <<filter-by-time>>.
. Filter the deployment data to display in the charts, as needed, to focus on areas of interest.
+
For guidance, see <<filter-by-data>>.

> . To get additional metrics on a specific app or API:

Reviewer: ```suggestion
. To get additional metrics for a specific app or API:
```

.. Find and click its name in *Entities List*.
+

> Apps in the *Started* state and APIs in the *Active* state are listed.

Reviewer: Make active? The list includes....

Author: Trying to put the most important bit to scan first.

.. Click *View Entity Metrics*, for example:
+

> image::insights-entity-clicked.png["Overview of a deployment metrics"]

Reviewer: singular or plural?

Author: plural -- there are several different charts

Reviewer: Remove "a"?

Author: ooooh, thx! 😆

Author: ```suggestion
image::insights-entity-clicked.png["Overview of deployment metrics"]
```

.. Review metrics in the charts for the entity, for example: 
+
[[charts-detailed]]
image::insights-deployment-charts.png["Dashboard for a deployed app or API"]

[calloutlist]
. Details page for a Mule app or API implementation
. Charts with detailed metrics
+
Detailed metrics are available in dashboards specific to <<entity-metrics-apps, Mule apps>> and <<entity-metrics-apis, API implementations>>.
. Selector for the time range covered by the charts 
+

> By default, the date period in dashboards with detailed metrics on specific entities matches the time period selected for cross-entity charts in the *Anypoint Insights* overview. 

Reviewer: Is bold needed here? UI element?

Author: That's the name of the page in the UI. It doesn't have a label other than *Anypoint Insights*. It's also the name of the feature as it appears in the UI, so it wouldn't get translated anyway.

. Selector for auto-refresh interval, displayed when auto-refresh is enabled
+

> By default, the auto-refresh interval in dashboards with detailed metrics on specific entities matches the time period interval for cross-entity charts in the *Anypoint Insights* overview. 

Reviewer: bold?

Author: same

//
//H2: FILTER DATA
[[filters]]
== Filter Monitoring Data Across Deployments

You can narrow the data in *Anypoint Insights* charts by deployment details, such as the deployment type, and by time range.

* <<filter-by-data>>
* <<filter-by-time>>

//  H3: FILTER DEPLOYMENT DATA
[[filter-by-data]]
=== Filter Deployment Data

Use one or more filters to narrow the data in your charts. The filters support operators `=` and `IN`. Multiple filter settings are allowed. For example, this filter returns Mule apps deployed to CloudHub 2.0: 

----
entity_type = 'APP' deployment_type = 'CLOUDHUB2'
---- 

This table lists supported filters:

[%header,cols="10a,80a,10a"]
|===
| Filter | Operators | Description

|`deployment_type`
|`=`, `IN`

> |Filter by deployment target, such as CloudHub 2.0 (*CLOUDHUB2*) or Runtime Fabric (*RTF*), for example:

Reviewer: Bold because UI element?

Author: Yes, both are options in the UI.

----
deployment_type IN ['CLOUDHUB2', 'RTF']
----

|`entity_type`
|`=`, `IN`
|Filter by deployment type (*APP* or *API*), for example:
----
entity_type = 'APP'`
----

|`entity`
|`=`, `IN`
|Filter by specific app or API deployments, for example:
----
entity IN ['99aea888-77de-6e06-5c55-d1234de56789', '09f99999-888d-7b77-6da5-bebd43210fc1']
----

|===

`IN` captures exact matches to any comma-separated element in the array.

//  H3: FILTER TIME PERIOD
[[filter-by-time]]
=== Filter Data by Time Period

> Use the time period selector (<<dashboard-overview, item 6>> in the example) to display data points collected for a given time period (such as the last 15 minutes or the last 24 hours). Your setting applies across all data in *Anypoint Insights* overview, including the *Entities List*.

Reviewer: ```suggestion
Use the time-period selector (<<dashboard-overview, item 6>> in the example) to show data points collected for a given time period (such as the last 15 minutes or the last 24 hours). Your setting applies across all data in *Anypoint Insights* overview, including items in the *Entities List*.
```

+
image::insights-environment-selector.png["Environment selector in Anypoint Monitoring"]
* To select a default environment:

.. Click your profile icon (see <<profile, item 2>> in the example). 
.. Select the name of your profile to open Access Management *Settings*.
.. In *Settings*, select an environment as the default environment to view for the selected business group in Anypoint Monitoring, for example: 
+
image::insights-default-selector.png["Default selector in Access Management"]
.. Return to Anypoint Monitoring to check for the new business group setting.
+
Refresh the page, if necessary. 

//
//H2: CHANGE BUSINESS GROUP
[[change-bg]]
== Change the Business Group

Charts in Anypoint Insights display metrics for deployments in a specific business group. If you belong to more than one business group, you can change the selected group and the business group that Anypoint Monitoring selects by default.

//do not change the order of numbers in the graphic or calloutlist
[[profile]]
image::insights-business-group-change.png["Business group selection options"]

[calloutlist]

> . Business group selector 

Reviewer: ```suggestion
. Business-group selector 
```

Reviewer: I am now doubting my compound-adjective edits...

Author: well, too late -- I'm going with it 😆

. User profile, for selecting defaults

> To change the business group setting: 

Reviewer: ```suggestion
To change the business-group setting: 
```

. Open Anypoint Monitoring. 
. Set a new business group:

> * To select a business group without changing the default, use the business group selector in Anypoint Insights. 

Reviewer: ```suggestion
* To select a business group without changing the default, use the business-group selector in Anypoint Insights. 
```

* To select a default business group:

.. Click your profile icon (see <<profile, item 2>> in the example). 
.. Select the name of your profile to open Access Management *Settings*.
.. In Access Management *Settings*, select a business group to use as the default, for example:
+
image::insights-default-selector.png["Default selector in Access Management"]

> .. Return to Anypoint Monitoring to check for the new default setting.

Reviewer: Consider using partials for 227 and 229.

Author: Adding to [W-17361464](https://gus.lightning.force.com/a07EE000026JsovYAC)

Author: Followup: 
* 227 isn't repeated anywhere at this time. 

* 229 is repeated, so an include is warranted.

+
Refresh the page, if necessary. 

[[download-chart-data]]
== Download Chart Data

After opening a set of charts in Anypoint Insights, you can download metrics from charts that provide a download menu in the chart.

> . Click the vertical ellipsis on a chart to open the download menu. 

Reviewer: Good place for an actual icon, maybe?

Author: Oh yeah, I creaeted one that's used elsewhere. Adding. 
TODO: @dukesphere

Author: ```suggestion
. Click the image:more-menu-vert-ellipsis.png["",15,15] (*More menu*)) on a chart to open the download menu. 
```

. Select the file format for your metrics:

* *Download as CSV*
* *Download as XLS*

//H2: REFERENCES
== Insights Chart Reference

Understand the monitoring data in Anypoint Insights charts:

* <<charts-performance>>
* <<entities-list>>
* <<entity-metrics-apps>>
* <<entity-metrics-apis>>

include::partial$include-entities.adoc[tags="entity-definition"]

//H3: PERFORMANCE
[[charts-performance]]
=== Performance Metrics and Trends

> *Performance Metrics* and *Performance Trends* provide insights across your entities:

Reviewer: UI elements?

Author: Yep. With the exception of those subscription tables in index.adoc, all bold I added was for UI elements.

include::partial$include-insights-charts.adoc[tags="data-scope"]

Total Entities::
Total number of entities

Entities with Errors::
Total number of entities that have received errors

* For API implementations, this metric counts HTTP status codes that identify errors (for example, `404` or `502`). 
* For Mule apps, this metric counts response codes that indicate an error (for example, `FAILED`).

Total Requests::
Total number of requests to the listed entities

Error Rate:: 
For entities with request errors, the percentage equal to the total number of request errors divided by the total number of requests
+
For example, assume that 1 out of 5 total requests to _API A_ produces an error and that 2 requests out of 5 to _App B_ produces an error. The error rate is 3/10, or 30%. 

//
// H4: Performance Trends
[[performance-trends-overview]]
==== Performance Trends

Monitor high-level performance over time across your entities. 
include::partial$include-entities.adoc[tags="entity-definition"]

include::partial$include-insights-charts.adoc[tags="data-scope"]

Entities with Slow Response Times::
Response times, in milliseconds, of the entities with the slowest response times
+

> The metric uses percentile values for response times (calculated from highest to lowest times) to identify entities with the slowest responses. You can select the *p99* (default), *p95*, *p90*, or *p75*.

Reviewer: ```suggestion
The metric uses percentile values for response times (calculated from highest to lowest times) to identify entities with the slowest responses. You can select *p99* (default), *p95*, *p90*, or *p75*.
```

Error Volume:: 
Number of errors received by the entities over time

> * For API implementations, this metric counts HTTP error codes (for example, `404` or `502`). 

Reviewer: opportunity for reuse?

Author: I"ll check for dups via
[W-17361464](https://gus.lightning.force.com/a07EE000026JsovYAC)

* For Mule apps, this metric counts error response codes (for example, `FAILED`).

Request Volume:: 
Number requests over time for the entities 

//
//H3: ENTITIES LIST
[[entities-list]]
=== Entities List

> Check key metrics on individual Mule apps and APIs (_entity_), and click an entity's name to get many detailed charts on the entity. 

Reviewer: ```suggestion
Check key metrics on individual Mule apps and APIs (_entity_), and click an entity's name to get detailed charts for the entity. 
```

*Entities List* provides these metrics:

Entity Name:: 
Name of the entity

//NOTE: changed value Api -> API
Entity Type:: 
Type of entity, such as *API* or *Application* (Mule app)

Request Volume::
Total number of requests to this entity

Response Time (p99):: 
Response time, in milliseconds, for the 99th percentile of requests to this entity
+
This response time is greater than 99% of all response times for this entity.

Error Rate:: 
Percentage of requests to this entity that resulted in errors 

Deployment Type:: 
Deployment target of this entity, such as *CloudHub 2.0* or *Runtime Fabric* 

//
//H3: ENTITY METRICS for APPS (ONLY)
[[entity-metrics-apps]]
=== Application Details

Anypoint Monitoring provides details on individual Mule apps deployed to a supported target.

* <<entity-metrics-app-header>>
* <<entity-metrics-app-overview>>
* <<entity-metrics-app-requests>>
* <<entity-metrics-app-errors>>
* <<entity-metrics-app-jvm>>

include::partial$include-entities.adoc[tags="entity-definition"]

[[entity-metrics-app-header]]
==== Mule Application Metrics

> High-level metrics are available in Application Details dashboards for individual Mule apps. 

Reviewer: Should Application Details be bold (UI element) or lowercase (generic)?

Author: hmm, need to double check. Don't remember why capitalized.

Author: ```suggestion
High-level metrics are available in *Application Details* dashboards for individual Mule apps. 
```

Author: I left out the bold accidentally, it seems. The name appears in Insights UI as a part of a breadcrumb.

include::partial$include-insights-charts.adoc[tag="app-detail-requests"]

[[entity-metrics-app-errors]]
==== Error Metrics for a Mule Application

Charts with error metrics are available in the *Errors* tab for a Mule app in Anypoint Insights.

include::partial$include-insights-charts.adoc[tags="data-scope"]

include::partial$include-insights-charts.adoc[tag="app-detail-errors"]

[[entity-metrics-app-jvm]]
==== JVM Metrics for a Mule Application

Charts with JVM metrics are available in the *JVM* tab for a Mule app in Anypoint Insights.

include::partial$include-insights-charts.adoc[tags="data-scope"]

include::partial$include-insights-charts.adoc[tag="app-detail-jvm"]

//
//H3: ENTITY METRICS for APIs (ONLY)
[[entity-metrics-apis]]
=== API Details

Anypoint Monitoring provides details on individual API implementations deployed to a supported target.

* <<entity-metrics-api-header>>
* <<entity-metrics-api-overview>>
* <<entity-metrics-api-requests>>
* <<entity-metrics-api-errors>>
* <<entity-metrics-api-performance>>
* <<entity-metrics-api-client-apps>>

include::partial$include-entities.adoc[tags="entity-definition"]

[[entity-metrics-api-header]]
==== API Implementation Metrics

> High-level metrics are available in *API Details* dashboards for individual API implementations.

Reviewer: bold here.

Author: UI

[[entity-metrics-api-requests]]
==== Request Metrics for an API Implementation 

Charts with request metrics are available in the *Requests* tab for each API implementation that you can monitor in Anypoint Insights.

include::partial$include-insights-charts.adoc[tags="data-scope"]

include::partial$include-insights-charts.adoc[tags="api-detail-requests"]

[[entity-metrics-api-errors]]
==== Error Metrics for an API Implementation 

Charts with error metrics are available in the *Errors* tab for each API implementation that you can monitor in Anypoint Insights.

include::partial$include-insights-charts.adoc[tags="data-scope"]

include::partial$include-insights-charts.adoc[tags="api-detail-errors"]

[[entity-metrics-api-performance]]
==== Performance Metrics for an API Implementation

Charts with performance metrics are available in the *Performance* tab for each API implementation that you can monitor in Anypoint Insights.

include::partial$include-insights-charts.adoc[tags="data-scope"]

include::partial$include-insights-charts.adoc[tags="api-detail-performance"]

[[entity-metrics-api-client-apps]]
==== Client Application Metrics for an API Implementation 

> Charts with client Application metrics are available in the *Client Applications* tab for each API implementation that you can monitor in Anypoint Insights. 

Reviewer: ```suggestion
Charts with client application metrics are available in the *Client Applications* tab for each API implementation that you can monitor in Anypoint Insights. 
```

include::partial$include-insights-charts.adoc[tags="data-scope"]

include::partial$include-insights-charts.adoc[tags="api-detail-client-apps"]

//
//H2: DATA COLLECTION
[[data-collection]]
== Data Collection Intervals in Anypoint Monitoring Charts

Anypoint Monitoring charts display data in configurable date-time ranges, which include the last 15 minutes, hour, week, a custom date-time range, and other periods.

* <<automatic-interval>>
* <<data-collection-intervals>>

[[automatic-interval]]
=== Data Points in Anypoint Monitoring Charts

Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. 

//TBD: PROPOSAL
Maximum granularity of the data captured in charts:

> * 1 minute for metrics captured within the last 7 days

Reviewer: Technically, we spell out 1-9, but I'm not sure here.

Author: I can change. That style strikes me as less scannable than using the number in cases like this. Not sure what the clear distinction would be, but when talking about metrics in charts, numbers make more sense to me. ¯\_(ツ)_/¯

Reviewer: Yeah, I feel similarly.
Salesforce style is very complicated: https://confluence.internal.salesforce.com/x/4ASGC

Reviewer: I guess it falls under time: time including years, months, weeks, and days

Author: I saw this to spell out "If the number is smaller than 10 and isn't a measurement"
This is a metric, so it falls under "measurement". Keeping that precious "1". 🤣

Author: And the UI uses numbers for this stuff, so it matches the experience better.
